imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.java

```diff
@@ -439,6 +443,10 @@ protected synchronized boolean updateDecodeJob(EncodedImage encodedImage, boolea
       return ret;
     }
 
+    private boolean isNotPartialData(EncodedImage encodedImage) {
+      return encodedImage.getEncodedCacheKey() != null;
```

Comment:
I'm not sure this is a reliable way to tell whether this image is partial or not. Can you show why you think this is correct, or use `isLast` as I mentioned above?